test/test.js

```diff
@@ -7,6 +7,14 @@ describe('resolve-dependency-path', function() {
     assert(resolved.indexOf(__dirname) === 0);
   });
 
+  it('resolves with multiple periods in the dependency path', function() {
+    var depPath = './bar.baz.qux';
```

Comment:
Good q. It gets to the else block in the case of ./bar.baz.qux because path.extname always returns an empty string the string has a period as the first character (which is true in this case). http://nodejs.org/api/path.html#path_path_extname_p